### PR TITLE
Bump Sphinx to 4.5.0

### LINKED ISSUE #15778: The highlighted search terms in the online refman are annoying
#### Description of the problem

The highlighted text can be very annoying, for example: https://coq.inria.fr/refman/user-extensions/syntax-extensions.html?highlight=notation#coq:cmd.Notation

Note that the URL itself mentions the highlighting, and manually editing the URL gets rid of it. I wish there was a better way though.

We identified two possible avenues towards getting rid of it in [this zulip discussion](https://coq.zulipchat.com/#narrow/stream/237977-Coq-users/topic/refman.3A.20avoiding.20highlighting.20by.20default.3F):

1. Include a button to hide the highlighting, like [this](http://2.python-requests.org/en/master/user/quickstart/?highlight=payload#passing-parameters-in-urls). Sphinx has this feature, but not Read the Docs, and it seems it will continue not having it:
  https://github.com/readthedocs/sphinx_rtd_theme/issues/816
  See also https://github.com/readthedocs/sphinx_rtd_theme/pull/876
This hypothetical button could perhaps be implemented on Coq's side.

2. Deploy the refman with Sphinx >= 4.5.0 in order to take advantage of https://github.com/sphinx-doc/sphinx/pull/9337. Note that 4.5.0 is the dev version at the time of writing.

cc. @cpitclaudel

### PR DESCRIPTION
**Pull request by ana-borges**
To obtain `/` as a quick search shortcut and `Esc` as a shortcut to remove search highlighting.
Sphinx 4.5.0 was released in March 2022.

Fixes / closes #15778

- ~~[ ] Added / updated **test-suite**.~~

- [x] Added **changelog**.
- [x] Added / updated **documentation**.
  
  - ~~[ ] Documented any new / changed **user messages**.~~
  - ~~[ ] Updated **documented syntax** by running `make doc_gram_rsts`.~~

- ~~[ ] Opened **overlay** pull requests.~~

<!-- Pointers to relevant developer documentation:

Contributing guide: https://github.com/coq/coq/blob/master/CONTRIBUTING.md

Test-suite: https://github.com/coq/coq/blob/master/test-suite/README.md

Changelog: https://github.com/coq/coq/blob/master/doc/changelog/README.md

Building the doc: https://github.com/coq/coq/blob/master/doc/README.md
Sphinx: https://github.com/coq/coq/blob/master/doc/sphinx/README.rst
doc_gram: https://github.com/coq/coq/blob/master/doc/tools/docgram/README.md

Overlays: https://github.com/coq/coq/blob/master/dev/ci/user-overlays/README.md

### CODE REVIEW

**Review by Zimmi48 — APPROVED** [review]
Thanks!
Just FWIW, Debian's new stable version, that was released earlier this month, has Sphinx 5.3.0 packaged.

File: doc/changelog/11-standard-library/17772-bump-sphinx.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+- **Changed:**
+  Sphinx 4.5.0 or above is now required to build the reference manual, so now /
```
[the remark this answers is not in the document]
Comment (ana-borges) [reply doc/changelog/11-standard-library/17772-bump-sphinx.rst:2]: Right, I think this was just inattention on my part.

**Comment by Zimmi48** [comment]
On my computer, when testing the refman artifact generated from this PR, it looks like the search feature is broken (never returns).

**Comment by Zimmi48** [comment]
FWIW, testing locally with Sphinx 5.1.1 (the version currently accessible from `nix-shell`) gives me a search that terminates (returns results) but with no search summary:

image

**Comment by ana-borges** [comment]
This was a much deeper rabbit hole than expected. Here are some findings:

- Inspecting a local copy of the docs does not necessarily behave like inspecting a version generated by CI due to the url being of the form `/home/bla/...` instead of `http://bla/...`.
In particular, `Search` doesn't give any results if using the former url in my machine, although it does in Théo's. However, it doesn't show summaries of each result, as shown above.
In order to get around both problems we can run a local server, for example with `python3 -m http.server`

- Finding combinations of python packages that work together is not always trivial. Some combinations of versions, specially very old + very recent are accepted by `pip` but lead to build failures for the doc.

File: doc/README.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ To produce the complete documentation in HTML, you will need Coq dependencies
 listed in [`INSTALL.md`](../INSTALL.md). Additionally, the Sphinx-based
 reference manual requires Python 3, and the following Python packages:
 
-  - sphinx >= 3.0.2
+  - sphinx >= 4.5.0
   - sphinx_rtd_theme >= 0.4.3
```
Comment (Zimmi48) [inline doc/README.md:34]:
```suggestion
  - sphinx_rtd_theme >= 1.0.0
```

**Comment by Zimmi48** [comment]
@coqbot merge now

**Pull request merged**